pkg/logpoller/service.go

```diff
@@ -31,21 +31,30 @@ type service struct {
 	client  ton.APIClientWrapped // TON blockchain client
 	filters FilterStore          // Registry of active filters
 	loader  TxLoader             // Transaction loader returning loaded txs
-	indexer TxIndexer            // Transaction indexer returning indexed logs
+	parser  TxParser             // Transaction parser returning logs
 	store   LogStore             // Log storage (MVP: in-memory, to be replaced with ORM)
 
 	pollPeriod         time.Duration // How often to poll for new blocks
 	lastProcessedBlock uint32        // Last processed masterchain sequence number
 }
 
+type ServiceOptions struct {
+	Config   Config // TODO: use global relayer config
```

Comment:
Don't agree with this TODO. You should have a LogPooler specific Config here, and then compose the Relayer total config using this specific one + others for other components (e.g., TXM).